core/message.h

```diff
@@ -49,4 +49,24 @@ static inline int uint64_to_bin(uint8_t *ptr, int size, uint64_t val, int be) {
   }
 }
 
+// move data from *ptr to *val. set "be" if *ptr stores big endian data
+static inline int bin_to_uint64(uint64_t *val, uint8_t *ptr, int size, bool be) {
+  *val = 0;
+
```

Comment:
`size` should be checked here. In `uint64_to_bin()` the size check is done implicitly, but not in `bin_to_uint64()`